holoviews/core/data/__init__.py

```diff
@@ -354,13 +373,52 @@ def __getitem__(self, slices):
         return data
 
 
-    def sample(self, samples=[]):
+    def sample(self, samples=[], closest=True, **kwargs):
         """
         Allows sampling of Dataset as an iterator of coordinates
         matching the key dimensions, returning a new object containing
-        just the selected samples.
-        """
-        return self.clone(self.interface.sample(self, samples))
+        just the selected samples. Alternatively may supply kwargs
+        to sample a coordinate on an object.
+        """
+        if kwargs and samples:
+            raise Exception('Supply explicit list of samples or kwargs, not both.')
+        elif kwargs:
+            sample = [slice(None) for _ in range(self.ndims)]
+            for dim, val in kwargs.items():
+                sample[self.get_dimension_index(dim)] = val
+            samples = [tuple(sample)]
+
+        # Note: Special handling sampling of gridded 2D data as Curve
+        # may be replaced wih more general handling
+        # see https://github.com/ioam/holoviews/issues/1173
+        from ...element import Table, Curve
+        if len(samples) == 1:
+            sel = {kd.name: s for kd, s in zip(self.kdims, samples[0])}
+            dims = [kd for kd, v in sel.items() if not np.isscalar(v)]
+            selection = self.select(**sel)
+            new_type = Table
+            kdims = self.kdims
+            if self.interface.gridded and self.ndims == 2 and len(dims) == 1:
+                new_type = Curve
+                kdims = [self.get_dimension(kd) for kd in dims]
+                selection = tuple(selection.columns(kdims+self.vdims).values())
+            elif np.isscalar(selection):
+                selection = [samples[0]+(selection,)]
+            else:
+                selection = tuple(selection.columns(kdims+self.vdims).values())
+            return self.clone(selection, kdims=kdims, new_type=new_type)
```

Comment:
Given the above comment (can return Curve early) then you could do...

```python
            selection = (tuple(selection.columns(kdims+self.vdims).values())
                         if np.isscalar(selection) else  [samples[0]+(selection,)])
```

That might be a little long for your taste so I'll leave it up to you.